frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDetailsActions.tsx

```diff
@@ -78,10 +108,33 @@ const PipelineRunDetailsActions: React.FC<PipelineRunDetailsActionsProps> = ({ o
               ) : (
                 <React.Fragment key="compare-runs" />
               ),
-              <DropdownSeparator key="separator" />,
-              <DropdownItem key="delete-run" onClick={() => onDelete()}>
-                Delete
-              </DropdownItem>,
+              !isRunActive ? (
+                !isExperimentActive ? (
+                  <Tooltip
+                    position="left"
+                    content={
+                      <div>
+                        Archived runs cannot be restored until its associated experiment is
+                        restored.
+                      </div>
+                    }
+                  >
+                    {RestoreDropdownItem}
+                  </Tooltip>
+                ) : (
+                  RestoreDropdownItem
+                )
+              ) : (
+                <React.Fragment key="restore-run" />
+              ),
+              !isRunActive ? (
+                <React.Fragment key="delete-run">
+                  <DropdownSeparator key="separator" />
+                  <DropdownItem onClick={() => onDelete()}>Delete</DropdownItem>
+                </React.Fragment>
+              ) : (
+                <React.Fragment key="delete-run" />
```

Comment:
this looks like it should be archive not none

https://www.figma.com/proto/etbN0ccCcZR1qKCpJXQNAg/Pipeline-phase-2?page-id=615%3A34927&node-id=1585-95721&viewport=3206%2C1251%2C0.65&t=oARGr6aU5d4T3KjF-1&scaling=min-zoom&starting-point-node-id=1585%3A95719

Reply:
Updated